platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblemBuilder.java

```diff
@@ -61,29 +53,34 @@ public class DefaultProblemBuilder implements InternalProblemBuilder {
     private DocLink docLink;
     private List<String> solutions;
     private Throwable exception;
-    //TODO Reinhold make private again
     private AdditionalData additionalData;
     private boolean collectStackLocation = false;
-    private final AdditionalDataBuilderFactory additionalDataBuilderFactory;
-    private final Instantiator instantiator;
-    private final PayloadSerializer payloadSerializer;
     private ProblemDiagnostics diagnostics;
 
-    public DefaultProblemBuilder(AdditionalDataBuilderFactory additionalDataBuilderFactory, Instantiator instantiator, PayloadSerializer payloadSerializer) {
-        this.additionalDataBuilderFactory = additionalDataBuilderFactory;
-        this.instantiator = instantiator;
-        this.payloadSerializer = payloadSerializer;
+    public DefaultProblemBuilder(
+        ProblemsInfrastructure infrastructure
+    ) {
+        this.problemsInfrastructure = infrastructure;
         this.additionalData = null;
         this.solutions = new ArrayList<String>();
     }
 
-    public DefaultProblemBuilder(@Nullable ProblemStream problemStream, AdditionalDataBuilderFactory additionalDataBuilderFactory, Instantiator instantiator, PayloadSerializer payloadSerializer) {
-        this(additionalDataBuilderFactory, instantiator, payloadSerializer);
-        this.problemStream = problemStream;
+    public DefaultProblemBuilder(
+        @Nullable ProblemStream problemStream,
+        AdditionalDataBuilderFactory additionalDataBuilderFactory,
+        Instantiator instantiator,
+        PayloadSerializer payloadSerializer,
+        IsolatableFactory isolatableFactory,
+        IsolatableToBytesSerializer isolatableSerializer
```

Comment:
❓ Can we get rid of this constructor and only use the 'infrastructure'-using variant?